### ISSUE TITLE
[RFC]Compensation of hours by agreement

### ISSUE THREAD

**Issue by lauradiaz22** [OCA/hr-attendance#199]
Good morning,

A common need has been detected in different sectors. It is the compensation of overtime and night hours worked. In certain jobs there are night shifts, and these hours are not equivalent to a standard working hour. The same can happen with overtime.

In Spain, it is the collective bargaining agreement that establishes the compensation for each type of hour over and above the standard hour. 

That is to say, if a worker works a night hour and the agreement establishes that it is worth 50% more than the ordinary hour, he would have to compensate 1.50, not 1 hour. And that information could be reflected in Odoo.

Regarding the last sentence, that would be the development proposal, to be able to establish by configuration, the compensation by type of hour depending on the agreement. And that both workers and managers can see how many hours they have accumulated to compensate.

Thanks in advance

**Comment by pedrobaeza** [comment]
Check https://www.youtube.com/watch?v=kwmdq5nRqS4 to see if it's covered with Pierre's developments.

**Comment by lauradiaz22** [comment]
Thank you very much for all the information.

I have been watching the video and testing the addon within the goose runboat, and I found it very complete but it lacks a very important detail, the weightings. You need to know the exact hours that can be compensated or that go on payroll, and for that you have to apply the weighting established by the agreement. Within the development, it does not allow you to add this weighting, it collects the hours as they are registered as if it were an ordinary hour. Would it be a good option to add this functionality to the module and migrate it to 16? (we need it in 16).

**Comment by pedrobaeza** [comment]
Yes, I think it can be something to be added to the existing modules instead of going with something from scratch. Note that Odoo also has some overtime mechanism in core, so maybe you can plug something into it too.

**Comment by lauradiaz22** [comment]
Thank you for your help. We will take it into consideration and explore the best way to integrate it into the addon.

**Comment by adasatorres** [comment]
Good morning, I have added this functionality during the migration (#203) to version 16.0 of the hr_attendance_hour_type_report addon, as it already includes a report with nighttime hours.

**Comment by rrebollo** [comment]
@lauradiaz22 were you finally able to address the topic with OCA?
